water_neuron/src/conversion.rs

```
@@ -15,6 +20,81 @@ use icrc_ledger_types::icrc2::transfer_from::TransferFromArgs;
+        Some(withdrawal_request) => {
+            let caller_account = Account {
+                owner: caller,
+                subaccount: None,
+            };
+            if caller_account == withdrawal_request.receiver {
+                withdrawal_request.icp_due
+            } else {
+                return Err(CancelWithdrawalError::BadCaller {
+                    message: format!("Caller did not match owner."),
+                });
+            }
+        }
+        None => {
+            return Err(CancelWithdrawalError::RequestNotFound);
+        }
+    };
+
+    let icp_stake_e8s = icp_due
+        .checked_sub(ICP::from_e8s(2 * DEFAULT_LEDGER_FEE))
+        .expect("ICP due should always be more than 10.");
+
+    stop_dissolvement(neuron_id).await.unwrap();
+    match merge_neuron(SIX_MONTHS_NEURON_NONCE, neuron_id)
+        .await
+        .unwrap()
+        .command
+        .expect("Command should always be set.")
+    {
+        CommandResponse::Merge(response) => {
+            mutate_state(|s| {
+                if s.neuron_id_to_withdrawal_id.get(&neuron_id).is_some() {
+                    process_event(
+                        s,
+                        EventType::MergeNeuron {
+                            icp_stake_e8s,
+                            receiver: Account {
+                                owner: caller,
+                                subaccount: None,
+                            },
+                            neuron_id,
+                        },
+                    );
+                    schedule_now(TaskType::ProcessPendingTransfers);
+                    schedule_now(TaskType::ProcessLogic);
+                }
+            });
+            Ok(*response)
+        }
+        CommandResponse::Error(e) => Err(CancelWithdrawalError::GovernanceError(e)),
+        _ => Err(CancelWithdrawalError::BadCommand {
+            message: format!("Expected merge commande got other."),
+        }),
```

Here you are losing some information, you should return a better error.

On side note `commande` -> `command`